winlogbeat/eventlog/poll_windows.go

```diff
@@ -0,0 +1,72 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build windows
+// +build windows
+
+package eventlog
+
+import (
+	"errors"
+	"io"
+
+	"golang.org/x/sys/windows"
+
+	"github.com/elastic/beats/v7/winlogbeat/sys"
+	"github.com/elastic/beats/v7/winlogbeat/sys/winevent"
+	win "github.com/elastic/beats/v7/winlogbeat/sys/wineventlog"
+)
+
+// oldestEvent returns the oldest event held in the system event log.
+func oldestEvent(work []byte, buf *sys.ByteBuffer) (winevent.Event, error) {
+	event, err := windows.CreateEvent(nil, 0, 0, nil)
+	if err != nil {
+		return winevent.Event{}, err
+	}
+	defer windows.CloseHandle(event)
+
+	s, err := win.Subscribe(0, event, "", "", 0, win.EvtSubscribeStartAtOldestRecord)
```

Comment:
I think this might not be doing what I would expect. This should be attempting to get the newest event's record ID to compare with reader's current event ID. I'm not sure if this can be done exclusively through flags or if an `EvtSeek` is needed.

Reply:
Yes, you are right. I was off-by-sign.

Reply:
I think this is right now; the docs are weak, but they say that the `EvtSeek`d handle should have come from a call to `EvtQuery`.

Reply:
It's not right.